clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1946,7 +1979,7 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
   }
 
   bool handlePointerType(FieldDecl *FD, QualType FieldTy) final {
-    addParam(FD, FieldTy, SYCLIntegrationHeader::kind_pointer);
+    addParam(FD, FieldTy, SYCLIntegrationHeader::kind_std_layout);
```

Comment:
Not sure that RT will be fine with this change...

Reply:
Yea.. It looks like there are several test failures. Little surprising because nothing failed locally.

Reply:
If you don't have OpenCL in your paths, check-sycl will run only on host device, which doesn't care about device compilation.

Reply:
I thought I did set up the runtimes before testing.. Let me check again. I thought the runtime won't have an issue because addParam just calculates sizes and offsets and I thought that remains the same when the pointer is wrapped inside a struct as well, since it is the only field. Maybe I was mistaken. I didn't really consider alignment. @erichkeane thoughts?

Reply:
There might be another runtime handling for `kind_pointer` value. I'm not sure that we need to change it in the compiler.

Reply:
I would think it would handle that, yes...